addons/test_website/static/tests/tours/website_settings.js

```diff
@@ -0,0 +1,45 @@
+/** @odoo-module */
+
+import tour from "web_tour.tour";
+
+const websiteName = "Website Test 2";
+
+tour.register("website_settings_m2o_dirty", {
+    test: true,
+    url: "/web",
+},
+[
+    tour.stepUtils.showAppsMenuItem(),
+    {
+        content: "open settings",
+        trigger: ".o_app[data-menu-xmlid='base.menu_administration'",
+    }, {
+        content: "open website settings",
+        trigger: ".settings_tab .tab[data-key='website']",
+    }, {
+        content: "Open website switcher",
+        trigger: "input#website_id",
+    }, {
+        content: `Select ${websiteName} in the website switcher`,
+        trigger: `li:has(.dropdown-item:contains('${websiteName}'))`,
+    }, {
+        content: "Open website switcher again",
+        trigger: "input#website_id",
+    }, {
+        content: "click on the fake website setting after checking the edited website",
+        trigger: "button[name='action_website_test_setting']",
+        extra_trigger: `.dropdown a.dropdown-item.ui-state-active:contains('${websiteName}')`,
```

Comment:
> You should at least check that the previous test worked, eg wait for the website 2 name to be loaded in the input, here I am pretty sure the tour is clicking on your action before the onchange rpc even returned

@rdeodoo Unfortunately it's quite complicated to check that an input is filled with something in a step as it's not in the HTML (you can't do a css selector on a **property**). So I've found a solution which consists of checking that the correct website is active in the website switcher. It's not ideal but I don't see any other clean way. Maybe I'll have a better idea this weekend :thinking:

Reply:
> Unfortunately it's quite complicated to check that an input is filled with something in a step as it's not in the HTML (you can't do a css selector on a property)

Simply work around it: enable the "Shared Customer Accounts" on website 2, disable it on website 1 (to be sure) and add in the next step `div[name=shared_user_account] input[type=checkbox]:checked'`.

> So I've found a solution which consists of checking that the correct website is active in the website switcher

Isn't that already true the moment you select website two, before the rpc is fired (and so returned)?